flink-table/flink-table-planner-blink/src/test/java/org/apache/flink/table/planner/factories/TestFileFactory.java

```diff
@@ -198,7 +196,9 @@ public void close() throws IOException {
 
         @Override
         public TypeInformation<RowData> getProducedType() {
-            return InternalTypeInfo.ofFields(DataTypes.STRING().getLogicalType());
+            // For ScanTableSource the output type is determined by planner,
+            // the result of this method would not be used.
```

Comment:
Nit: update the comment to mention the purpose of returning null is validating the output type can be handled by planner correctly.